src/pyramid_jwt/__init__.py

```diff
@@ -41,6 +46,45 @@ def create_jwt_authentication_policy(config, private_key=None, public_key=None,
             audience=audience)
 
 
+def _request_create_token(request, principal, expiration=None,
+                          audience=None, **claims):
+
+    return request.authentication_policy.create_token(
+        principal, expiration, audience, **claims
+    )
+
+
+def _request_claims(request):
+    return request.authentication_policy.get_claims(request)
+
+
+def _configure(config, auth_policy):
+    config.set_authentication_policy(auth_policy)
+    config.add_request_method(lambda request: auth_policy,
+                              'authentication_policy', reify=True)
```

Comment:
I'm not in favour of adding a `authentication_policy` attribute to the request, for several reasons:

- the JWT authentication policy may not be the main one, but one of several combined using something like pyramid_authstack. That means the name might be misleading.
- theoretically there can be multiple JWT authentication policies in use at the same time, and this will result in a conflict in that (admittedly unlikely) scenario.
- using a public attribute (i.e. without a `_` prefix) for what is essentially an implementation detail feels wrong to me. If you need to stick private data somewhere you can also use the registry as a dictionary.

Using a locale closure and `request_create_token` / `request_claims` functions like the current code feels like a simpler and cleaner approach.

Reply:
Hi. All these sound like cases to be dealt by `pyramid_authstack`. Although Pyramid's registry does allow for multiple implementations to the same interface, it's authentication layer only supports one authentication policy per app (so much that you had to create an auth stack layer to use multiple)

IMO, as a standalone lib, `pyramid_jwt` should note have to deal with the fact that **there may be** stack in place.

That being said, this is merely a shorthand, convenient way, to obtain the current authentication policy in place. It may as well be obtained querying the registry, so it's up to you. I can remove it if you prefer.